ir/function.cpp

```diff
@@ -236,6 +305,9 @@ void Function::instr_iterator::operator++(void) {
   if (++II != IE)
     return;
   ++BBI;
+  while (BBI != BBE && (*BBI)->empty()) {
+    ++BBI;
+  }
```

Comment:
Hey @nunoplopes and @aqjune As I mentioned in https://github.com/AliveToolkit/alive2/issues/540 , the instr iterator for function is crashing when source has empty basicblocks (for #sink), these three lines make the iterator "jump over" the empty block. After this adding these three lines, alive won't crash any more for transforms with #sink block.  Since this is changing some basic stuff for alive, are you okay with this?